internal/resource/http.go

```diff
@@ -247,3 +250,20 @@ func (c HttpClient) getReaderWithHeader(url string, header http.Header) (io.Read
 		}
 	}
 }
+
+// addSystemInfoHeaders will add an ID, a VERSION_ID (if present), and a
+// VARIANT_ID (if present) header to the client's ignition config GET request
+func addSystemInfoHeaders(header *http.Header) {
+	or, err := osrelease.New()
+	if err != nil {
+		return
```

Comment:
This worries me a little. I read it as "if we couldn't get the osrelease information just silently skip it even though we were asked to do it". Is that the expected result?

Reply:
Crawford said that if one of these field are not present, just skip it. Should I instead add the 3 headers no matter if the field is present or not? So if the length is 0, send an empty string as the value.

Reply:
I see. In that case I'd recommend one or more of the following:

- Change the name to something like `addSystemInfoHeadersIfAvailable`
- Add a comment to the function noting that if no osrelease data is found then this function is a noop
- log a warning or info when no osrelease data is found

Reply:
I'd suggest that we bubble-up the error here, and let the caller check for that and log a non-critical entry in that case.

Reply:
What @lucab suggests is what I'd normally expect from a function like this.

```go
func addSystemInfoHeaders(header *http.Header) error {
    or, err := osrelease.New()
    if err != nil {
        return err
    }
[...]
    return nil
}
```

Reply:
@ashcrow I feel that the first two points are unnecessary/already taken care of by the current comment. Also the function will create a ID header, since this isn't an optional field so there should never be a case where this function NOOP.

Reply:
So what I originally had, but instead of returning err in getReaderWithHeader() log the error right?

Reply:
Fixed